week4/src/components/History.tsx

```diff
@@ -17,12 +17,12 @@ export default function History({
 }: iHistoryProps) {
   const navigate = useNavigate();
 
-  const selectHistory = (_history: string | undefined) => {
+  const selectHistory = (_history: string) => {
     navigate(`/search/${_history}`);
     setIsOpen(false);
   };
 
-  const deleteHistory = (_history: string | undefined) => {
+  const deleteHistory = (_history: string) => {
     setHistory((prev: Array<string>) =>
       prev.filter((item) => item !== _history)
     );
```

Comment:
이거 위애 
```ts
 history: string[];
```
로 사용되고 있어서! 

```suggestion ts
setHistory((prev) => prev.filter((item) => item !== _history));
```

타입지정안해죠도 데요! 중복 지정은 불필요해서요! 모든걸 다 타입지정하는게 마냥 조은건 아닐슈두 있다는거!

Reply:
오홍 그렇네! 저렇게 하면 중복 지정이 되는구나! 하긴 중복으로 굳이 또 타입을 선언해줄 이유가 없긴 하겠다! 고마워~!